### ISSUE TITLE
Error with nested connect under connectOrCreate

### ISSUE THREAD

**Issue by darrellwarde** [neo4j/graphql#840]
**Describe the bug**

Error with nested connect under connectOrCreate - this is pulled from Trello.

**Type definitions**

```gql
type Fact {
    id: String @unique
    votes: [Vote!]! @relationship(type: "VOTE_HAS_TOPICS", direction: IN)
}
           
type Vote {
    id: String @unique
    facts: [Fact!]! @relationship(type: "VOTE_HAS_FACT", direction: OUT)
    topics: [Topic!]! @relationship(type: "VOTE_HAS_TOPICS", direction: OUT)
}

type Topic {
    name: String @unique
}
```

**To Reproduce**

1. Execute the following Mutation:

```gql
mutation {
    updateFacts(
        where: { id: "fact1" }
        connectOrCreate: {
            votes: {
                where: { node: { id: "lmnop" } }
                onCreate: {
                    node: {
                        id: "lmnop"
                        topics: { connect: { where: { node: { name: "my-topic" } } } }
                    }
                }
            }
        }
    ) {
       facts {
            id
            votes {
                id
            }
        }
    }
}
```

2. See the error:

```
Neo4jError: Property values can only be of primitive types or arrays thereof. Encountered: Map{connect -> List{Map{where -> Map{node -> Map{name -> String("my-topic")}}}}}
```

**Expected behavior**

Error is not thrown.

**Additional context**

This is pulled from Trello, which also had the information: "Is thrown, due to params not being properly serialized"

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @darrellwarde. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

Thanks again! :pray:

**Comment by angrykoala** [comment]
Error confirmed with the example provided.

Note that the database must already contain the following data to reproduce the error:
```cypher
CREATE(t:Topic {name: "topic1"})
CREATE(v:Vote {id: "vote1"})
CREATE(f:Fact {id: "fact1"})

MERGE(v)-[:VOTE_HAS_FACT]->(f)
MERGE(v)-[:VOTE_HAS_TOPICS]->(t)
```

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @darrellwarde! :pray: We will now prioritise the bug and address it appropriately.

**Comment by darrellwarde** [comment]
@angrykoala, I honestly can't remember, does `connectOrCreate` even support nested operations under `onCreate`? I could have sworn that this wouldn't work because of the syntax abilities of `MERGE... ON CREATE`?

**Comment by angrykoala** [comment]
@darrellwarde You are right, currently, it is not supported, and it may not be feasible to implement this at the moment.
However, I think a problem still exists in that the GraphQL types support this construct so we may need to re-scope this bug in changing the types, so our GraphQL typedefs are accurate. Thoughts?

**Comment by darrellwarde** [comment]
> @darrellwarde You are right, currently, it is not supported, and it may not be feasible to implement this at the moment. However, I think a problem still exists in that the GraphQL types support this construct so we may need to re-scope this bug in changing the types, so our GraphQL typedefs are accurate. Thoughts?

Precisely, this came up with my conversation with @Andy2003 that `create` and `connectOrCreate.onCreate` use the same input type, when in fact the one for `onCreate` can only accept primitive types.

We should get this in for 3.0.0 as technically its a breaking change to the schema (even though it currently doesn't work).

**Comment by darrellwarde** [comment]
See #872

**Comment by angrykoala** [comment]
The we should close this one in favour of #872
